Tests/UnitTests/Update-UnityPackageManagerConfig.Tests.ps1

```diff
@@ -0,0 +1,119 @@
+BeforeAll {
+    Import-Module (Join-Path $PSScriptRoot '..\..\UnitySetup\UnitySetup.psd1') -Force
+
+    Mock -CommandName 'Import-UnityProjectManifest' -MockWith { @() } -ModuleName UnitySetup
+    Mock -CommandName 'Import-TOMLFile' -MockWith { @() } -ModuleName UnitySetup
+    Mock -CommandName 'Update-PackageAuthConfig' -MockWith { @() } -ModuleName UnitySetup
+    Mock -CommandName 'Export-UPMConfig' -MockWith { @() } -ModuleName UnitySetup
+    Mock -CommandName 'Invoke-WebRequest' -MockWith {
+        return [pscustomobject]@{
+            StatusCode = 200
+            Content    = '{"patToken": {"token": "mockedPATToken"}}'
+        }
+    } -ModuleName UnitySetup
+    Mock -CommandName 'Confirm-PAT' -MockWith { $true } -ModuleName UnitySetup
+}
+
+Describe 'Update-UnityPackageManagerConfig' {
+
+    Context 'Input Validation' {
+        It 'throws on empty parameters' {
+            { Update-UnityPackageManagerConfig } | Should -Throw "*insufficient number of parameters were provided."
+        }
+
+        It 'throws on null ProjectManifestPath' {
+            { Update-UnityPackageManagerConfig -ProjectManifestPath "" } | Should -Throw "*The argument is null or empty.*"
+        }
+    }
+
+    Context 'Function Execution' {
+        It 'supports a root folder target' {
+            { Update-UnityPackageManagerConfig -SearchPath ".\Data\" -SearchDepth 5 } | Should -Not -Throw
```

Comment:
Can we modify these relative paths to use $PSScriptRoot? Right now if I don't cd into the 'Tests' folder then I get errors running the unittests.ps1 script.

Reply:
Good call - will update this